File: Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSub,currSub,i=1;
        maxSub = currSub = nums[0];
        int last = nums.size();
        while(i < last){
            currSub = max(currSub+nums[i],nums[i]);
            maxSub = max(maxSub,currSub);
            i++;
        }
        return maxSub;
    }
};

File: Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        maxSub = nums[0]
        currSub = nums[0]
        length = len(nums)
        for i in range(1,length):
            currSub = max(currSub+nums[i],nums[i])
            maxSub = max(maxSub,currSub)
        return maxSub
      
